feat(landoscript): respect closed tree for android l10n sync

bhearsum opened the pull request:
I originally thought this was only used for l10n-bump, but it turns out we need to check treestatus for android l10n sync tasks as well.

bhearsum committed:
feat(landoscript): respect closed tree for android l10n sync

I originally thought this was only used for l10n-bump, but it turns out we need to check treestatus for android l10n sync tasks as well.

I also added a test for the "dontbuild is only respected for l10n-bump" codepath while I'm touching this area.

jcristau reviewed (approved):
Do we somewhere test that "approval required" is treated as "open" for our purposes?  (Beta is rarely if ever actually "open")

bhearsum committed:
fix(landoscript): test treestatus for both open and approval required status

I've switched to "approval required" as the default state (because that's what we typically see for beta, where it's actually checked), while maintaining a test for the "open" state as well.
